### ISSUE TITLE
Prevent initial error spam for required attributes

### ISSUE THREAD

**Issue by majkinetor** [json-editor/json-editor#856]
I have a form with 20 or so questions, all are required. When it is created, there is a red spam that all questions are missing. If option `"show_errors": "never"` is set, all disappear after single one is edited.

How can I prevent this spam ? I want error to show on field change without initial spam:

Here is the gif showing 3 required questions that disappear as soon first is edited.
![json ](https://user-images.githubusercontent.com/85767/97982026-739e6f00-1dd3-11eb-9ce3-1c01ea87d14d.gif)

As a side note, this div should probably have some class `errors` to be easily moved with selectors. Currently the one must get it by position which may be flaky.

**Comment by majkinetor** [comment]
This thing with errors shown when errors should not be shown is probably a bug.

**Comment by schmunk42** [comment]
Could you post an example schema?

**Comment by majkinetor** [comment]
Here is the one from https://github.com/json-editor/json-editor/issues/343#issuecomment-721983461

[Json Editor Proof](
https://pmk65.github.io/jedemov2/dist/demo.html?schema=EQbwOgdgBFbAJAZwMYAsCmBbAhnAXLMKgC7EAOeA9JQFaID2EAtChjgHT0BOA5pQCZdsAM2JMADABZKrLNgDEcADSQYcfumEBLCFuJbGifFHDQYhYugAexY6fPm4xAJ5l0xuImJcdPZaodCYW4cW2ACJ2tibC50XGAVMwc4WOx%2BRgAbZ2NhbAzEdETAwkwdABl0CB5iVGMAJiLAuHoyfUM7AOK4Hh9%2BAH1kegyAV0wII3CoADZGruAdMmHiPowtHhIPYABGAFZYzDhO8wBfI%2BPZwh1gjqTHYBc3Tav6f1u1YBa28Zvi956tfqDEZjCYEGZHZLAf79ABGqQA1sZvMN3G8oKdbucjnAAI5MSw2H5NYDBLihTbQljeLTIMIXd5kLgtdBcfToUEmCF3BZLIm%2FQifAzfSb2fl3QaYMjYWlIrgorkOLFoyHwWLCTbyASaHR6IWISgEukKu6M5ms5wAeS4Gi4xi24mNSv5cGefN%2BcFVmg1Wu0ui%2B%2BtdCWNDKZbnNVptdvEDuV6K5GMVASdFj0GXckzgAGZ2FBAPgggD4QQAsIIAGEEAYiCADhAoPmoIBBEEAQiCAARBi3XAKwggAkQQBcIPnABogUAbLcA3CAtwBMIMXALwgdbH%2BfrlfLhZb3ZbUA7xcA%2FCDluvFgdTwDMIFBVw2ey2O5vXu8HumIh8YTR0DKg2Y4KTyRmob0qT5741mq09W71FoiDYDCaZ9Og%2FB6H0dCMMYMYwMmKToDiwxaLE%2FDGAA2ti2xnncdSHGYAC637AKaYZshyorvFsbrvJ66qvpqGi%2BrqhiUHihq4ZC%2BjEGmmzZlsUCALggxbDi2hYbkuK5dlAgDaIEuB5HiudZQOWNaroAwiBjhulYVoAmiAjku5aAOwgwlluW%2BZjluu59sWxnDj2NmrgO47FkoVaFlONZTlAVkVpZ1lLvWTatju5aLlxJqhiyFG0dyEDXCKwaEBoKA%2BL%2BMGvgAPFoAB8WVkLlub9oWA75nuimLiexZQI2LaAIwgNUlvu%2BYroAbCBLs18m1kFikORJdaVhpqkTi2%2FY1sNq75nVjVQMeln5oAPCA1l2e6VjW%2BYVt21nlbZ9k9jVVm%2BeZAXbktqkDpWZUrmdi2%2BeWDkaTmgAoIA2bWiWOCn5pdg0di27krtWbZ1tWfkWVZ26Vk2%2B3dtp%2B4uWO7kzU1NX1RphbDVZYnzp9VbDvO7mLg2UCLi1M0%2Bfm9XQ2VbU6cuxYyd1lXHs2UCFvmB7lp9g1QIAWiDloAoiD5g5d3TnWo4Ng9xa809WWUIVBVFYeVWs4t%2BYdap6kabzonNutxZqyT4njlA%2BktT95YrhdlZqxr5aLXrUDGTVZtqZpOujrzS5ifJe5UwdpvlgZo45oACCAyaTIO1gOGmAOIgxbdu5Najh2Nv5k2bOo%2BjS5Y%2BJLZjuwcsK5QeUEfyCaBBX8EXHA%2BFJWiHpqt6zE6v67H4lEkXnqml6ENmdRQIAGCCLc5LaLfnm5bTVwu7cz1VsxzktebOWse9Lw2U6JLbO6ppnj4ASiBacWh6eYLUAaTuM8Z%2BtU3QxpDtQIABCAtpp9WXX9UCAIQgDb1eFLZ4F3QgZEYpaHZHFd4gYCBUXdMAVKyB0pfE2DlfKhURJiRnJnKAl0eqqRHkOceE5zLdnOnPTcmt3a627BvRqYljJGT3i2Q%2BVkT7aQFufS%2BDlr61lvnZe%2BNVn6v3fkub%2Bv9Fw5kAHgg64ayO2ktWbq9tR4EMnuFGqpCaoaUGqWR2R03ba11l7KsNDt41RMlAA%2BR8WHlQABSCwAJTsKviY7h6deEOxzEXFBuU0EtgwcnJcF8r5GWcXfceqlDFbzocuHs9sapM2VizFS7NObc3LIXeWnjADEIB2EKMkOZqUds1bBv9z41iHBJIcLVSniXLKWfxXZ37VhrApeJylfJLn7FuEyGlSaVkGqE3GujcbmOYUWbSd0AmcPOuWbB6kXHGXvudDRw4Gz73zkuTetClwCI0m%2FS2wif5%2FwkfVba3k6bDS7HdQ8HZhpNKrLHY83TJrTjKTUoJXZtpGz5oLYWNVobjybGk4updgCOnjGcJMkBjjACAA%3D%3D%3D&value=ETI%3D&code=EQehAIBUAsFNwGYHsA2KkHcCWA7A5uCrrAM7gCGATvALbk4Am5ALkpQJ4WPjXkNI4U7AHTgAmkgCu4AMb0KDBrMklWNWUgbxyAIyQA3bdx2x0GYQB0cWBOAAUAKxKwGWVpXC5V9GbCS3sRkxhACkAZQB5ADkAUVd3AEpwJxc3NmEtVUokdjsEgG4rFPi2cABecBxYDHBA%2FnNw6Li0yjt%2BGUkaWBxmYQBHSVgOMNNYGXc7C2AAYicBAFpU93nkShophIAaZJcWcgLgIA%3D&style=EQbQJghgLhC0DOBjAFgUwLYQA7WQKgF4AdYTAJwGsSBdAAgG9aAjCRCgczIHsBXAOzAAuWgGIAZgHZUANhkBuWuXYBLPrChcswgIwYFOMGFXt1mwbABMGWgF9gQA&theme=spectre&iconlib=spectre&object_layout=grid&template=default&show_errors=never&required_by_default=1&no_additional_properties=0&display_required_only=0&remove_empty_properties=1&keep_oneof_values=1&ajax=0&ajaxCredentials=0&show_opt_in=0&disable_edit_json=1&disable_collapse=1&disable_properties=1&disable_array_add=0&disable_array_reorder=1&disable_array_delete=0&enable_array_copy=0&array_controls_top=1&disable_array_delete_all_rows=1&disable_array_delete_last_row=1&prompt_before_delete=0&lib_aceeditor=0&lib_autocomplete=0&lib_sceditor=0&lib_simplemde=1&lib_select2=0&lib_selectize=0&lib_choices=0&lib_flatpickr=0&lib_signaturepad=0&lib_mathjs=0&lib_cleavejs=0&lib_jodit=0&lib_jquery=0&lib_dompurify=1)

with `show_errors=never` set. Enter single value and all errors will disappear.

**Comment by schmunk42** [comment]
You have `remove_empty_properties` set, is this intentional?

**Comment by majkinetor** [comment]
Yes, without them there are no validate errors, because empty values are considered enough for required (see my comment here https://github.com/json-editor/json-editor/issues/854#issuecomment-722036593)

**Comment by potop** [comment]
These behavior is really annoying.
We do use remove_empty_properties set for our business logic and now we have to hide all validation messages with css

**Comment by schmunk42** [comment]
PRs (with test) welcome ;)

**Comment by chandanbn** [comment]
Not deleting the required empty properties here: https://github.com/json-editor/json-editor/blob/master/src/editors/object.js#L1090 seems to do the trick.

replace 

```
if (isEmpty(result[key])) {
```

with 

```
if (isEmpty(result[key]) && !(Array.isArray(this.schema.required)) && this.schema.required.includes(key)) {
```

I feel changing any option should not make JSON output invalid (wrt schema), if it was valid before setting that option.
